### PR TITLE
WIP Plugin refactor

### PR DESCRIPTION
**Pull request by arekkas**
@inyono this is how we could refactor the plugins. what do you think?

### CODE REVIEW

**Comment by arekkas** [comment]
However, this is a rather large refactor because the plugins are used in a lot of places. alternatives are discussed in #252

**Comment by arekkas** [comment]
Also, what's missing is the default values which are defined in LayoutPlugin / ContentPlugin and I'm not sure if there's an easy / sane way to inject those because we might deal with react classes here.

File: examples/home/src/plugins/video/index.js [hunk 1]
```diff
@@ -11,47 +11,51 @@ type Props = {
...
-            <div className="navbar-header">
-              <a className="navbar-brand" href="http://ory.am">ORY</a>
-            </div>
-            <div id="navbar" className="navbar-right">
-              <ul className="nav navbar-nav">
-                <li><a href="https://github.com/ory-am/editor">GitHub</a></li>
-              </ul>
+class Video extends Component {
+  render() {
+    const { children } = this.prpos
```
Comment (inyono) [inline examples/home/src/plugins/video/index.js]: typo

**Comment by arekkas** [comment]
Rejected!

**Pull request closed without merging**